ONSdigital/census-rm-case-processor
# Fix Spring retries bug with transaction manager

# Motivation and Context
If we hit a problem when processing a Rabbit message, we aren't rolling back and handling retries properly - we are sending extra (duplicate) Rabbit messages, amongst other problems. This is quite fundamental to the reliable functioning of RM, so we can trust our code not to corrupt any data in the event of exceptions.

# What has changed
Added a `ChainedTransactionManager` per the suggestion of Dr David Syer in his article _Distributed transactions in Spring, with and without XA_ (http://www.infoworld.com/article/2077963/distributed-transactions-in-spring--with-and-without-xa.html)

# How to test?
It's super hard.

Suggestion: set the cache fetch count to 3,000 or more, and the timeout to 5 seconds or less, then run the AT scenario `Tranche 2 household case details to be sent to the Field Work Management Tool` with a freshly booted up Case Processor - you would see duplicate cases being emitted, erroneously, in the original code.

# Links
Trello: https://trello.com/c/J4uL5FRY
